src/clients/c++/request_common.cc

```diff
@@ -925,12 +925,29 @@ InferContextImpl::GetReadyAsyncRequest(
     return Error(
         RequestStatusCode::UNAVAILABLE,
         "No asynchronous requests have been sent");
+  } else {
+    bool has_dangling_request = false;
+    for (auto& ongoing_async_request : this->ongoing_async_requests_) {
+      auto request_impl =
+          static_cast<RequestImpl*>(ongoing_async_request.second.get());
+      if (!request_impl->HasCallback()) {
+        has_dangling_request = true;
+        break;
+      }
+    }
+    if (!has_dangling_request) {
+      return Error(
+          RequestStatusCode::UNAVAILABLE,
+          "No asynchronous requests should be returned, all requests will be "
```

Comment:
"No asynchronous requests can be returned, all outstanding requests will signal completion via their callback function"